examples/opentelemetry-example-app/src/opentelemetry_example_app/context_propagation_example.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module serves as an example for baggage, which exists
+to pass application-defined key-value pairs from service to service.
+"""
+# import opentelemetry.ext.http_requests
+# from opentelemetry.ext.wsgi import OpenTelemetryMiddleware
+
+import flask
+from flask import request
+import requests
+
+from opentelemetry import propagation, trace
+from opentelemetry.distributedcontext import CorrelationContextManager
+from opentelemetry.sdk.context.propagation import b3_format
+from opentelemetry.sdk.trace import Tracer
+from opentelemetry.sdk.trace.export import (
+    BatchExportSpanProcessor,
+    ConsoleSpanExporter,
+)
+from opentelemetry.baggage import BaggageManager
+
+
+def configure_opentelemetry(flask_app: flask.Flask):
+    trace.set_preferred_tracer_implementation(lambda T: Tracer())
+
+    # Global initialization
+    (baggage_extractor, baggage_injector) = BaggageManager.http_propagator()
+    (b3_extractor, b3_injector) = b3_format.http_propagator()
+    # propagation.set_http_extractors([b3_extractor, baggage_extractor])
+    # propagation.set_http_injectors([b3_injector, baggage_injector])
+    propagation.set_http_extractors([b3_extractor])
+    propagation.set_http_injectors([b3_injector])
+
+    # opentelemetry.ext.http_requests.enable(trace.tracer())
+    # flask_app.wsgi_app = OpenTelemetryMiddleware(flask_app.wsgi_app)
+
+
+def fetch_from_service_b() -> str:
+    # Inject the contexts to be propagated. Note that there is no direct
+    # reference to tracing or baggage.
+    headers = {"Accept": "application/json"}
+    propagation.inject(headers)
+    print(headers)
+    resp = requests.get("https://opentelemetry.io", headers=headers)
+    return resp.text
+
+
+def fetch_from_service_c() -> str:
+    # Inject the contexts to be propagated. Note that there is no direct
+    # reference to tracing or baggage.
+    headers = {"Accept": "application/json"}
+    propagation.inject(headers)
+    print(headers)
+    resp = requests.get("https://opentelemetry.io", headers=headers)
+    return resp.text
+
+
+app = flask.Flask(__name__)
+
+
+@app.route("/")
+def hello():
+    tracer = trace.tracer()
+    tracer.add_span_processor(BatchExportSpanProcessor(ConsoleSpanExporter()))
+    with propagation.extract(request.headers):
```

Comment:
right, this is the same as the "Simpler API" example for anyone that doesn't need to use the context explicitly. As I suspect this would be the more common case, I like how much simpler it is.